MigrationTool/Program.cs

```diff
@@ -0,0 +1,279 @@
+                    AddKeyWithLog(playerExtraOptionsPanelIni, "chkBoxForceRandomColors",    "Location", "12,50");
+                    AddKeyWithLog(playerExtraOptionsPanelIni, "chkBoxForceRandomTeams",     "Location", "12,72");
+                    AddKeyWithLog(playerExtraOptionsPanelIni, "chkBoxForceRandomStarts",    "Location", "12,94");
+                    AddKeyWithLog(playerExtraOptionsPanelIni, "chkBoxUseTeamStartMappings", "Location", "12,130");
+                    AddKeyWithLog(playerExtraOptionsPanelIni, "btnHelp",                    "Location", "160,130");
+                    AddKeyWithLog(playerExtraOptionsPanelIni, "lblPreset",                  "Location", "12,156");
+                    AddKeyWithLog(playerExtraOptionsPanelIni, "ddTeamStartMappingPreset",   "Location", "65,154");
+                    AddKeyWithLog(playerExtraOptionsPanelIni, "ddTeamStartMappingPreset",   "Size",     "157,21");
+                    AddKeyWithLog(playerExtraOptionsPanelIni, "teamStartMappingsPanel",     "Location", "12,189");
+
+                    continue;
+
+                case (Version.V_2_11_1): // https://github.com/CnCNet/xna-cncnet-client/releases/tag/2.11.1.0
+                    // Add ClientDefinitions.ini->[Settings]->RecommendedResolutions
+                    AddKeyWithLog(clientDefsIni, "Settings", "RecommendedResolutions", "1280x720");
+                    AddKeyWithLog(clientDefsIni, "Settings", "MaximumRenderWidth", "1280");
+                    AddKeyWithLog(clientDefsIni, "Settings", "MaximumRenderHeight", "720");
+                    clientDefsIni.WriteIniFile();
+                    continue;
+
+                case (Version.V_2_11_2): // https://github.com/CnCNet/xna-cncnet-client/releases/tag/2.11.2.0
+                    // Remove ClientUpdater.xml and SecondStageUpdater.xml
+                    var listExtraXMLs = new List<string>(2) { "ClientUpdater.xml", "SecondStageUpdater.xml" };
+                    Log("Remove ClientUpdater.xml and SecondStageUpdater.xml");
+
+                    foreach (var extraXml in listExtraXMLs)
+                    {
+                        Directory.GetFiles(resouresDir.FullName, extraXml, SearchOption.AllDirectories)
+                            .ToList()
+                            .ForEach(elem => SafePath.DeleteFileIfExists(elem));
+                    }
+
+                    // Add ClientDefinitions.ini->[Settings]->ShowDevelopmentBuildWarnings
+                    AddKeyWithLog(clientDefsIni, "Settings", "ShowDevelopmentBuildWarnings", "true");
+                    clientDefsIni.WriteIniFile();
+                    continue;
+
+                case (Version.V_2_12_0): // https://github.com/CnCNet/xna-cncnet-client/releases/tag/2.12.0
+                    // Remove Rampastring.Tools from Resources directory (not recursive)
+                    Log("Remove Resources/Rampastring.Tools.* (* -- dll, pdb, xml)");
```

Comment:
Resources/Rampastring.Tools.dll prevents the client from launching since the ~2.8.0.0 chaos. Not from v2.12.0 only